File: CS/E430_MVC/Controllers/HomeController.cs
using DevExpress.Web.Mvc;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace E430_MVC.Controllers
{
    public class HomeController : Controller {
        // GET: Home
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial() {
            return PartialView(BatchEditRepository.GridData);
        }

       public JsonResult UpdateData(string data) {
            try {
                BatchEditRepository.UpdateData(JsonConvert.DeserializeObject<GridDataItem>(data));
                return Json("OK");
            }
            catch (Exception ex) {
                return Json(ex.Message);
            }
        }
    }
}

File: VB/E430_MVC/Controllers/HomeController.vb
Imports DevExpress.Web.Mvc
Imports Newtonsoft.Json
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace E430_MVC.Controllers
	Public Class HomeController
		Inherits Controller

		' GET: Home
		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			Return PartialView(BatchEditRepository.GridData)
		End Function

	   Public Function UpdateData(ByVal data As String) As JsonResult
			Try
				BatchEditRepository.UpdateData(JsonConvert.DeserializeObject(Of GridDataItem)(data))
				Return Json("OK")
			Catch ex As Exception
				Return Json(ex.Message)
			End Try
	   End Function
	End Class
End Namespace